src/04-extensions.js

```diff
@@ -0,0 +1,212 @@
+const exampleDinosaurData = require("../data/dinosaurs");
+const exampleTicketData = require("../data/tickets");
+const exampleRoomData = require("../data/rooms");
+
+/*
+Write a function that will organize all of the dinosaurs by their period key. 
+The function should return an object where the keys are the period values, and 
+the values are an array of IDs.
+*/
+
+/* Expected Result
+  {
+    "Late Jurassic": [
+      "YLtkN9R37",
+      "GGvO1X9Zeh",
+      "BFjjLjea-O",
+      "iOVNUcv-ww",
+      "V53DvdhV2A",
+    ],
+    "Late Cretaceous": [
+      "WHQcpcOj0G",
+      "GKl035EYKN",
+      "2GglUqKT0G",
+      "Pr6kc4Q_Xf",
+      "ft5Gs5izdq",
+      "wuL4ddBinQ",
+    ],
+    // ...
+  }
+*/
+
+function organizeDinosaursByPeriod(dinos, key) {
+  //input an array of objects
+  //output an object where key represents period and value reps an array of dinoIDs
+
+  //create helper function to camelCase
+  function camelCaseKey(keyString) {
```

Comment:
In general, it's not best practice to define functions inside of other functions. There are some good reasons to argue for or against this (as is the case with just about everything, programming-related), but for now I would avoid doing so.

If you want a way to indicate that a helper function is _only_ a helper function, you can prefix the function with an underscore, for example `function _camelCaseKey(keyString) { ...`. This does not do anything magical, but is a common convention for "private" functions.

Happy to talk more about this~